moderntribe/square-one
# Feature/sqone 429/lefthook git hooks

## What does this do/fix?

- Replaces husky hooks with https://github.com/evilmartians/lefthook to run git-hooks. I couldn't get a number of things working properly and this executes much faster and is much more configurable. See https://github.com/evilmartians/lefthook/blob/master/docs/full_guide.md for configuration options.
- Commits will be prefixed with the Jira ticket for better historical tracking purposes in projects. Example with this branch: `[SQONE-429] my original commit message`

## QA

- yarn install should activate the hooks automatically, if not you can run `lefthook install` or `lefthook install -a` to remove all existing hooks and replace with these ones.
- phpcs should run only on changed files in the core plugin/theme and the mu-plugins folder.
- branches without a Jira ticket should just commit as normal without adding the ticket prefix.

## Tests

Does this have tests?

- [x] No, this doesn't need tests because it changes how git-hooks are configured.



[SQONE-429]: https://moderntribe.atlassian.net/browse/SQONE-429
